robottelo/hosts.py

```diff
@@ -1288,3 +1288,21 @@ def update_setting(self, name, value):
         setting = self.api.Setting().search(query={'search': f'name="{name}"'})[0]
         setting.value = value
         setting.update({'value'})
+
+    def register_to_dogfood(self):
+        dogfood_canonical_hostname = settings.repos.dogfood_repo_host.partition('//')[2]
```

Comment:
this has been introduced #8812 , as mentioned in PR, it should have been moved to robottelo/hosts

the problem is not with registration but with [`install_katello_ca`](https://github.com/SatelliteQE/robottelo/blob/b041a65c47beb75898f52a1e549a1d04809a89d1/robottelo/hosts.py#L319)
we have to assure that the ca is installed, but the name of the package is not called dogfood but it's named dhcp-***.

There is another issue security issue mentioned by @ogajduse  in #8812 , if you are instered @Gauravtalreja1

Reply:
@tstrych Thanks for pointing that, and for elaborating here, will give it a try to that other issue after this PR.

Reply:
Did you try `f'rpm -Uvh {settings.repos.dogfood_repo_host}/pub/katello-ca-consumer-latest.noarch.rpm'`?

Reply:
@swadeley Yes, I tried this one, the same command is used here https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L317

In the case of dogfood, `dogfood_repo_host` is a CNAME (alias record) of fetched hostname, that's why we don't have any package named `katello-ca-consumer-<dogfood_repo_host>`, so fetching actual hostname for this package is required here.

Reply:
> Consider changing the format of this value in settings to just be the fqdn, so you don't have to take these steps to try and resolve a hostname separately.


DOGFOOD_REPO_HOST from conf/repos.yaml is only used in two tests, so its not too much work to move protocol out of `conf/repos.yaml`:
```
(robottelo-master) [sjw@t480s robottelo]$ grep -r -i -B1 DOGFOOD_REPO_HOST  tests/
tests/foreman/sys/test_katello_certs_check.py-        rhel_vm.execute(
tests/foreman/sys/test_katello_certs_check.py:            f'yum -y localinstall {settings.repos.dogfood_repo_host}'
--
tests/foreman/installer/test_infoblox.py-        'yum -y localinstall '
tests/foreman/installer/test_infoblox.py:        f'{settings.repos.dogfood_repo_host}/pub/katello-ca-consumer-latest.noarch.rpm'
```

but then this might need an update:

conf/subscription.yaml:  DOGFOOD_URL: "@format {{this.repos.dogfood_repo_host}}"

although no test is using that yet as far as I could tell.

Reply:
> the problem is not with registration but with [`install_katello_ca`](https://github.com/SatelliteQE/robottelo/blob/b041a65c47beb75898f52a1e549a1d04809a89d1/robottelo/hosts.py#L319)
> we have to assure that the ca is installed, but the name of the package is not called dogfood but it's named dhcp-***.

On the /pub page the package is called `katello-ca-consumer-latest.noarch.rpm`